File: apps/website/src/lib/editor/tracing.ts
import { ingestSpans, tracer } from '$lib/otel';

export function startFrameSpan(documentId?: string) {
  const span = tracer.startSpan('frame');
  if (!span.isRecording()) return null;
  if (documentId) span.setAttribute('document.id', documentId);
  return span;
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function drainTraces(fn: () => any) {
  const traces = fn();
  if (!traces || traces.length === 0) return;

  for (const spans of traces) {
    ingestSpans(spans);
  }
}

File: legacy/mobile/lib/screens/native_editor/controller/tracing.dart
import 'package:opentelemetry/api.dart' as otel_api;
import 'package:typie/otel.dart';

otel_api.Span? startFrameSpan({String? documentId}) {
  final span = tracer.startSpan('frame');
  if (!span.spanContext.isValid) {
    return null;
  }
  if (documentId != null) {
    span.setAttribute(otel_api.Attribute.fromString('document.id', documentId));
  }
  return span;
}
